File: backend/src/messages/room.rs
use crate::models::room::Room;
use serde::{Deserialize, Serialize};

#[derive(Deserialize)]
pub struct RoomCreateRequest {
    pub description: String,
    pub name: String,
    pub public: bool,
    pub url: String,
}

#[derive(Serialize)]
pub struct RoomGetAllResponse {
    pub rooms: Vec<Room>,
}

File: frontend/src/messages/room.ts
export interface RoomCreate {
    description?: string,
    name?: string,
    public: boolean,
    url?: string,
    [key: string]: boolean | string | undefined,
}

export interface RoomAllResponse {
    rooms: Array<RoomResponse>,
}

export interface RoomResponse {
    description: string,
    name: string,
    url: string,
}
